osgp/shared/osgp-dto/src/main/java/org/opensmartgridplatform/dto/valueobjects/smartmetering/CircuitSwitchedStatusDto.java

```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright 2020 Alliander N.V.
+ *
+ * <p>Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ *
+ * <p>http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.opensmartgridplatform.dto.valueobjects.smartmetering;
+
+public enum CircuitSwitchedStatusDto {
+  INACTIVE(0),
+  INCOMING_CALL(1),
+  ACTIVE(2),
+  RESERVED(3);
+
+  private final int index;
+
+  private CircuitSwitchedStatusDto(final int index) {
+    this.index = index;
+  }
+
+  public int getIndex() {
+    return this.index;
+  }
+
+  public static CircuitSwitchedStatusDto fromIndexValue(final int value) {
+    for (final CircuitSwitchedStatusDto status : CircuitSwitchedStatusDto.values()) {
+      if (status.index == value) {
+        return status;
+      }
+    }
+    return RESERVED;
```

Comment:
`RESERVED` is only appropriate for values 3..255, I'd suggest to start this method by throwing an `IllegalArgumentException` if `value < 0` or `value > 255`.

Reply:
Done